### PR TITLE
Fix Datasource Update to no User/Password

### PR DESCRIPTION
**Pull request by Maddin-619**
Until now it was not possible to update a datasource that has a database with a user and password to one without user and password. When clearing the user and passowrd input field from a already existing datasource and hit the submit button the user and password reappears in the input field. The only way was to create a new datasource without adding user and passowrd for anonymous loggin. So I chanced the SQL updated function such that all fields (also empty fileds) gets updated.

### CODE REVIEW

**Comment by torkelo** [comment]
This looks ok, think it makes all these calls obsolete:

https://github.com/grafana/grafana/blob/09cd173e9249d5e54cf3c304d2430f0390c68f45/pkg/services/sqlstore/datasource.go#L167

File: pkg/services/sqlstore/datasource.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func UpdateDataSource(cmd *m.UpdateDataSourceCommand) error {
 			updateSession = sess.Where("id=? and org_id=?", ds.Id, ds.OrgId)
 		}
 
-		affected, err := updateSession.Update(ds)
+		affected, err := updateSession.AllCols().Omit("created").Update(ds)
```
Comment (bergquist) [inline pkg/services/sqlstore/datasource.go]: Sweet! Didnt know about AllCols :) 

All Torkel mentioned the sess.UseBool calls should be removed. Confusing to solve the same problem twice.

**Review by bergquist — CHANGES_REQUESTED** [review]

File: pkg/services/sqlstore/datasource.go (hunk 1 above)
Comment (bergquist) [reply pkg/services/sqlstore/datasource.go]: Merging this once those removes have been removed.

Comment (bergquist) [reply pkg/services/sqlstore/datasource.go]: Nevermind. I made the change myself since its so small.

**Review by bergquist — APPROVED** [review]

**Comment by daniellee** [comment]
This introduced a serious bug I think for SecureJsonData fields as it overwrites the `secure_json_data` column with an empty object if the user clicks saves and test a second time after entering a new password. Investigating now and will open a new issue to revert this change.

**Comment by Maddin-619** [comment]
Sorry my bad. It was a bit confusing for me, because there are differences in the way datasources are configured. MySQL and InfluxDB store credentials in User, Password columns in plaintext. Azure Monitor and PostgreSQL stores the key/password encrypted in the secureJsonData. Maybe it's better to get rid of the plaintext password column completely and switch to secureJsonData for all Datasources to store the key/password. After that we can think about a generic solution, which makes it possible to update to no user and password.

**Comment by daniellee** [comment]
@Maddin-619 yes, we definitely should switch all the datasources to use secureJsonData. It is bad, legacy code that datasources have unencrypted passwords.

This was not an obvious bug as there are only two core datasources that use this pattern. We will definitely try and come up with a solution to solve the original issue.

**Pull request merged**